Modules/Tokens/src/Tokens.Infrastructure/Persistence/Repository/TokensRepository.cs

```diff
@@ -24,28 +24,24 @@ public TokensRepository(TokensDbContext dbContext)
         _readonlyTokensDbSet = dbContext.Tokens.AsNoTracking();
     }
 
-    public async Task<Token> Find(TokenId id)
+    public async Task<Token> Find(TokenId id, IdentityAddress? activeIdentity)
     {
         var getMetadata = _readonlyTokensDbSet
             .Where(Token.IsNotExpired)
+            .Where(Token.CanBeCollectedBy(activeIdentity))
             .FirstWithId(id);
 
         var token = await getMetadata ?? throw new NotFoundException(nameof(Token));
 
         return token;
     }
 
-    public async Task<DbPaginationResult<Token>> FindAllWithIds(IEnumerable<TokenId> ids, PaginationFilter paginationFilter, CancellationToken cancellationToken)
+    public async Task<DbPaginationResult<Token>> FindAllWithIds(IdentityAddress activeIdentity, IEnumerable<TokenId> ids, PaginationFilter paginationFilter, CancellationToken cancellationToken)
     {
-        return await Find(null, ids, paginationFilter, cancellationToken);
+        return await Find(activeIdentity, ids, paginationFilter, cancellationToken);
     }
 
-    public async Task<DbPaginationResult<Token>> FindAllOfOwner(IdentityAddress owner, PaginationFilter paginationFilter, CancellationToken cancellationToken)
-    {
-        return await Find(owner, Array.Empty<TokenId>(), paginationFilter, cancellationToken);
-    }
-
-    private async Task<DbPaginationResult<Token>> Find(IdentityAddress? owner, IEnumerable<TokenId> ids, PaginationFilter paginationFilter, CancellationToken cancellationToken)
+    private async Task<DbPaginationResult<Token>> Find(IdentityAddress? identityAddress, IEnumerable<TokenId> ids, PaginationFilter paginationFilter, CancellationToken cancellationToken)
```

Comment:
Pretty redundant to have this private `Find` method now that it's only called from the `FindAllWithIds` method.
I would just shift the code over.

Reply:
I asked Timo about that yesterday, the answer was that they should be kept separated as public method returns one and private method returns multiple tokens.

Reply:
Hmm? I think you misunderstood Daniel. This method can actually be inline, as the only placed it is used at is line 39.

Reply:
Yeah, my bad, completely misunderstood...

changes made